app/uk.gov.hmrc.helptosavestridefrontend/repo/SessionStore.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavestridefrontend.repo
+
+import cats.data.{EitherT, OptionT}
+import cats.instances.either._
+import cats.syntax.either._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.libs.json.{Json, Reads, Writes}
+import play.modules.reactivemongo.ReactiveMongoComponent
+import uk.gov.hmrc.cache.model.Id
+import uk.gov.hmrc.cache.repository.CacheMongoRepository
+import uk.gov.hmrc.helptosavestridefrontend.config.FrontendAppConfig
+import uk.gov.hmrc.helptosavestridefrontend.models.HtsSession
+import uk.gov.hmrc.helptosavestridefrontend.metrics.Metrics
+import uk.gov.hmrc.helptosavestridefrontend.util.{Logging, PagerDutyAlerting, Result, toFuture}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext
+
+@ImplementedBy(classOf[SessionStoreImpl])
+trait SessionStore {
+
+  def get(implicit reads: Reads[HtsSession], hc: HeaderCarrier): Result[Option[HtsSession]]
+
+  def store(body: HtsSession)(implicit writes: Writes[HtsSession], hc: HeaderCarrier): Result[Unit]
+
+  def delete(implicit hc: HeaderCarrier): Result[Unit]
+}
+
+@Singleton
+class SessionStoreImpl @Inject() (mongo:             ReactiveMongoComponent,
+                                  metrics:           Metrics,
+                                  pagerDutyAlerting: PagerDutyAlerting)(implicit appConfig: FrontendAppConfig, ec: ExecutionContext)
+  extends SessionStore {
+
+  private val expireAfterSeconds = appConfig.mongoSessionExpireAfter.toSeconds
+
+  private val cacheRepository = new CacheMongoRepository("stride-sessions", expireAfterSeconds)(mongo.mongoConnector.db, ec)
```

Comment:
notice the `stride-sessions` collection name -  we can't  reuse the same `sessions` we use incase of frontend ,
ask my why

Reply:
is this because of the platform's recommendation not to have different microservices talking to the same database or is there another reason for it as well?

Reply:
we can but why to create a new db while we have dumped everything related to help to save in the `help-to-save` db ? but if we want to follow `the platform's recommendation not to have different micro-services talking to the same database` then we can create a new db

Reply:
I am also happy to use `help-to-save-frontend` db incase of digital service

Reply:
fixed for stride and also for digital : https://github.com/hmrc/help-to-save-frontend/pull/737/files